src/main/java/com/graphaware/module/uuid/generator/SequenceIdGenerator.java

```diff
@@ -0,0 +1,34 @@
+package com.graphaware.module.uuid.generator;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Result;
+
+import com.graphaware.common.uuid.UuidGenerator;
+import com.graphaware.module.GraphDatabaseServiceAware;
+
+/**
+ * A complex implementation of UuidGenerator that makes use of the provided GraphDatabaseService to 
+ * generate a numerically increasing sequence through Cypher.
+ */
+public class SequenceIdGenerator implements UuidGenerator, GraphDatabaseServiceAware {
+
+	public static final String cypher = "MERGE (sequenceMetadata:SequenceMetadata) set sequenceMetadata.sequence = coalesce(sequenceMetadata.sequence, 0) + 1 return sequenceMetadata.sequence";
```

Comment:
Thanks for your feedback.

In this case, I am not sure I follow or understand your reply. As written, the Cypher does not make use of parameters because it is self contained and has everything it needs to correctly increment and query the next value in the sequence. If you can provide some more background or context to your feedback to help me understand what you'd like to see improved I'll be happy to make any needed changes.